src/main/vcpf4/usbd_cdc_vcp.c

```diff
@@ -34,18 +34,22 @@ LINE_CODING g_lc;
 extern __IO uint8_t USB_Tx_State;
 __IO uint32_t bDeviceState = UNCONNECTED; /* USB device status */
 
-/* These are external variables imported from CDC core to be used for IN
- transfer management. */
-extern uint8_t APP_Rx_Buffer[]; /* Write CDC received data in this buffer.
- These data will be sent over USB IN endpoint
- in the CDC core functions. */
+/* These are external variables imported from CDC core to be used for IN transfer management. */
+
+/* This is the buffer for data received from the MCU to APP (i.e. MCU TX, APP RX) */
+extern uint8_t APP_Rx_Buffer[]; 
 extern uint32_t APP_Rx_ptr_out;
-extern uint32_t APP_Rx_ptr_in; /* Increment this pointer or roll it back to
+
+/* Increment this buffer position or roll it back to
  start address when writing received data
  in the buffer APP_Rx_Buffer. */
-__IO uint32_t receiveLength=0;
+extern uint32_t APP_Rx_ptr_in; 
+__IO uint32_t receiveLength = 0;
 
-usbStruct_t usbData;
+
+static uint8_t APP_Tx_Buffer[USB_RX_BUFSIZE]; 
```

Comment:
I like that you are making the TX buffer similar to the RX buffer. Could you please comment each one which data it actually manages? I get all confused whether its from the host computer or being sent to the host computer.